protocol-units/bridge/contracts/src/NativeBridge.sol

```diff
@@ -100,25 +142,52 @@ contract NativeBridge is AccessControlUpgradeable, PausableUpgradeable, INativeB
         uint256 amount,
         uint256 nonce
     ) internal {
-        // _l2l1RateLimit(amount);
+        _rateLimitInbound(amount);
+        // Ensure the bridge transfer has not already been completed
+        require(nonce > 0, InvalidNonce());
+        require(idsToInboundNonces[bridgeTransferId] == 0, CompletedBridgeTransferId());
         // Ensure the bridge transfer ID is valid against the initiator, recipient, amount, and nonce
         require(
             bridgeTransferId == keccak256(abi.encodePacked(initiator, recipient, amount, nonce)),
             InvalidBridgeTransferId()
         );
-        // Ensure the bridge transfer has not already been completed
-        require(noncesToIncomingBridgeTransferIds[nonce] == bytes32(0x0), CompletedBridgeTransferId());
 
         // Store the nonce to bridge transfer ID
-        noncesToIncomingBridgeTransferIds[nonce] = bridgeTransferId;
+        idsToInboundNonces[bridgeTransferId] = nonce;
 
         // Transfer the MOVE tokens to the recipient
         if (!moveToken.transfer(recipient, amount)) revert MOVETransferFailed();
 
         emit BridgeTransferCompleted(bridgeTransferId, initiator, recipient, amount, nonce);
     }
 
+    function setInsuranceFund(address _insuranceFund) external onlyRole(DEFAULT_ADMIN_ROLE) {
+        insuranceFund = _insuranceFund;
+        emit InsuranceFundUpdated(_insuranceFund);
+    }
+
+    function setRiskDenominator(uint256 _riskDenominator) external onlyRole(DEFAULT_ADMIN_ROLE) {
+        require(_riskDenominator > MINIMUM_RISK_DENOMINATOR, InvalidRiskDenominator());
+        riskDenominator = _riskDenominator;
+        emit RiskDenominatorUpdated(_riskDenominator);
+    }
+
     function togglePause() external onlyRole(DEFAULT_ADMIN_ROLE) {
         paused() ? _pause() : _unpause();
+        emit PauseToggled(paused());
+    }
+
+    /**
+     * @dev Rate limits the inbound transfers based on the insurance fund and risk denominator
+     * @param amount The amount to rate limit
+     */
+
+    function _rateLimitInbound(uint256 amount) public {
+        uint256 day = block.timestamp / 1 days;
```

Comment:
For soundness, you should also require that the amount does not equal `max(uint256)`.

Reply:
This is assured by the total supply of MOVE not being able to reach max(uint256). tx will revert once amount is invalid

Reply:
How is that assured? For now, that is the case because of the way the MOVE token is minted, but it is not represented in this contract.